impl/core/src/main/java/io/serverlessworkflow/impl/executors/RunScriptExecutor.java

```diff
@@ -0,0 +1,259 @@
+    PYTHON("python");
+
+    private final String lang;
+
+    ScriptLanguage(String lang) {
+      this.lang = lang;
+    }
+
+    public String getLang() {
+      return lang;
+    }
+
+    public static boolean isSupported(String lang) {
+      for (ScriptLanguage l : ScriptLanguage.values()) {
+        if (l.getLang().equalsIgnoreCase(lang)) {
+          return true;
+        }
+      }
+      return false;
+    }
+  }
+
+  @FunctionalInterface
+  private interface ScriptExecutor {
+    WorkflowModel apply(WorkflowContext workflowContext, TaskContext taskContext);
+  }
+
+  private ScriptExecutor scriptExecutor;
+
+  @Override
+  public void init(RunScript taskConfiguration, WorkflowDefinition definition) {
+    ScriptUnion scriptUnion = taskConfiguration.getScript();
+    Script script = scriptUnion.get();
+    String language = script.getLanguage();
+    boolean isAwait = taskConfiguration.isAwait();
+
+    WorkflowApplication application = definition.application();
+    if (language == null || !ScriptLanguage.isSupported(language)) {
+      throw new IllegalArgumentException(
+          "Unsupported script language: "
+              + language
+              + ". Supported languages are: "
+              + Arrays.toString(
+                  Arrays.stream(ScriptLanguage.values()).map(ScriptLanguage::getLang).toArray()));
+    }
+
+    String lowerLang = language.toLowerCase();
+
+    scriptExecutor =
+        (workflowContext, taskContext) -> {
+          String source;
+          if (scriptUnion.getInlineScript() != null) {
+            source = scriptUnion.getInlineScript().getCode();
+          } else if (scriptUnion.getExternalScript() == null) {
+            throw new WorkflowException(
+                WorkflowError.runtime(
+                        taskContext, new IllegalStateException("No script source defined."))
+                    .build());
+          } else {
+            source =
+                definition
+                    .resourceLoader()
+                    .load(
+                        scriptUnion.getExternalScript().getSource(),
+                        ResourceLoaderUtils::readString,
+                        workflowContext,
+                        taskContext,
+                        taskContext.input());
+          }
+
+          Map<String, String> envs = new HashMap<>();
+          if (script.getEnvironment() != null) {
+            for (Map.Entry<String, Object> entry :
+                script.getEnvironment().getAdditionalProperties().entrySet()) {
+              String value =
+                  ExpressionUtils.isExpr(entry.getValue())
+                      ? WorkflowUtils.buildStringResolver(
+                              application,
+                              entry.getValue().toString(),
+                              taskContext.input().asJavaObject())
+                          .apply(workflowContext, taskContext, taskContext.input())
+                      : entry.getValue().toString();
```

Comment:
This is an example of functor I mention in the general comment. 
The idea is that you generate a Map<String,WorkflowValueResolver<String> during init, by using Workfloutils.buildStringFilter (WorkflowApplication appl, String expression), and at runtime you actually invoke the apply (passing the Workflowcontext, TaskContext and WorkflowModel passed to apply method)